docs/DataProviders.md

```diff
@@ -650,4 +586,263 @@ const UserProfile = ({ userId }) => {
+- you can [send file using multipart/form-data](#sending-files-in-multipartform-data) that allow you to send a whole resource and their files;
+- or you might want [to send files to a third party service](#sending-files-to-a-third-party-service) such as CDN;
+- etc.
+
+### Sending Files In Base64
+
+The following `dataprovider` extends an existing `dataprovider` to convert images passed to `dataProvider.update('posts')` into Base64 strings. The example leverages [`withLifecycleCallbacks`](#adding-lifecycle-callbacks) to modify the `dataProvider.update()` method for the `posts` resource only.
+
+```js
+import { withLifecycleCallbacks } from 'react-admin';
+import simpleRestProvider from 'ra-data-simple-rest';
+
+const dataProvider = withLifecycleCallbacks(simpleRestProvider('http://path.to.my.api/'), [
+    {
+        /**
+         * For posts update only, convert uploaded images to base 64 and attach them to
+         * the `picture` sent property, with `src` and `title` attributes.
+         */
+        resource: 'posts',
+        beforeUpdate: async (params, dataProvider) => {
+            // Freshly dropped pictures are File objects and must be converted to base64 strings
+            const newPictures = params.data.pictures.filter(
+                p => p.rawFile instanceof File
+            );
+            const formerPictures = params.data.pictures.filter(
+                p => !(p.rawFile instanceof File)
+            );
+
+            const base64Pictures = await Promise.all(
+                newPictures.map(convertFileToBase64)
+            )
+            const pictures = [
+                ...base64Pictures.map((dataUrl, index) => ({
+                    src: dataUrl,
+                    title: newPictures[index].name,
+                })),
+                ...formerPictures,
+            ];
+            return dataProvider.update(
+                resource,
+                { data: { ...params.data, pictures } }
+            );  
+        }
+    }
+]);
+
+/**
+ * Convert a `File` object returned by the upload input into a base 64 string.
+ * That's not the most optimized way to store images in production, but it's
+ * enough to illustrate the idea of dataprovider decoration.
+ */
+const convertFileToBase64 = file =>
+    new Promise((resolve, reject) => {
+        const reader = new FileReader();
+        reader.onload = () => resolve(reader.result);
+        reader.onerror = reject;
+        reader.readAsDataURL(file.rawFile);
+    });
+
+export default myDataProvider;
+```
+
+**Tip**: Use `beforeSave` instead of `beforeUpdate` to do the same for both create and update calls.
+
+### Sending Files In `multipart/form-data`
+
+In case you need to upload files, as you would with an HTML form but to your API, you can use [FormData](https://developer.mozilla.org/en-US/docs/Web/API/FormData) API that uses the same format a form would use if the encoding type were set to `multipart/form-data`.
```

Comment:
```suggestion
In case you need to upload files, as you would with an HTML form but to your API, you can use the [FormData](https://developer.mozilla.org/en-US/docs/Web/API/FormData) API . It uses the same format as a form would use if the encoding type were set to `multipart/form-data`.
```